src/components/ViewSnippetHeader.tsx

```diff
@@ -55,6 +55,23 @@ export default function ViewSnippetHeader() {
     )
   }
 
+  const [isStarred, setIsStarred] = useState(false)
+
+  useEffect(() => {
+    const fetchStarStatus = async () => {
+      if (!snippet) return
+      try {
+        const { data } = await axios.post("/snippets/get_star", {
+          snippet_id: snippet.snippet_id,
+        })
+        setIsStarred(data.is_starred)
+      } catch (error: any) {
+        console.error("Error fetching star status:", error)
+      }
+    }
+    fetchStarStatus()
+  }, [snippet])
```

Comment:
You need to add support for `snippet.is_starred`, the production api already has this so no other implementation is acceptable